File: Templates.py
import sys
input = sys.stdin.readline
def sol(arr, n):
    pass
 
t = int(input())
for case in range(t):
    n = int(input())
    arr = list(map(int,input().split()))
    s = input()
    print(sol(arr, n))

# Binary Index Tree
class BIT:
    def __init__(self, n):
        self.n = n
        self.nums = [0] * (n + 1)
        self.sums = [0] * (n + 1)

    def update(self, ind, val):
        ind += 1
        diff = self.nums[ind]-val
        self.nums[ind] = val
        while (ind <= self.n):
            self.sums[ind] -= diff
            ind += (ind & -ind)
        return True

    def query(self, ind):
        ind += 1
        total=0
        while (ind > 0):
            total += self.sums[ind]
            ind -= (ind & -ind)
        return total

    def sum(self, l, r):
        return self.query(r) - self.query(l-1)
    
# Disjoint Set
class DJS:
    def __init__(self, n):
        self.arr = [i for i in range(n)]
        self.rank = [1] * n
    
    def find(self, x):
        if x != self.arr[x]:
            self.arr[x] = self.find(self.arr[x])
        return self.arr[x]
    
    def union(self, x, y):
        xp = self.find(x)
        yp = self.find(y)
        if xp != yp:
            if self.rank[xp] >= self.rank[yp]:
                self.arr[yp] = xp
                self.rank[xp] += self.rank[yp]
            else:
                self.arr[xp] = yp
                self.rank[yp] += self.rank[xp]
            
            
            
class Segment_TreeNode():
    def __init__(self, lo, hi, s=0, mi=0, ma=0):
        self.lo = lo
        self.hi = hi
        self.sum = s
        self.min = mi
        self.max = ma
            
class Segment_Tree():
    def __init__(self, arr):
        self.n = len(arr)
        logs = 1
        tmp = self.n - 1
        while tmp:
            logs += 1
            tmp >>= 1
        
        self.Nodes = [0] * ((1 << logs) - 1)
        self.lazy = [0] * ((1 << logs) - 1)
        self.arr = arr
        self.build(0, 0, self.n - 1)
        
    def build(self, ind, lo, hi):
        if lo == hi:
            val = self.arr[lo]
            newNode = Segment_TreeNode(lo, hi, val, val, val)
            self.Nodes[ind] = newNode
            return
    
        mid = (lo + hi) // 2
        left = (ind << 1) + 1
        right = (ind << 1) + 2
        self.build(left, lo, mid)
        self.build(right, mid + 1, hi)
        
        nSum = self.Nodes[left].sum + self.Nodes[right].sum
        nMin = min(self.Nodes[left].min, self.Nodes[right].min)
        nMax = max(self.Nodes[left].max, self.Nodes[right].max)
        
        newNode = Segment_TreeNode(lo, hi, nSum, nMin, nMax)
        self.Nodes[ind] = newNode
        return
    def push_up(self, ind):
        self.Nodes[ind].sum = self.Nodes[(ind << 1) + 1].sum + self.Nodes[(ind << 1) + 2].sum
        self.Nodes[ind].min = min(self.Nodes[(ind << 1) + 1].min, self.Nodes[(ind << 1) + 2].min)
        self.Nodes[ind].max = max(self.Nodes[(ind << 1) + 1].max, self.Nodes[(ind << 1) + 2].max)
        
    def push_down(self, ind):
        if not self.lazy[ind]: return
        self.Nodes[(ind << 1) + 1].sum = self.Nodes[ind].max * (self.Nodes[(ind << 1) + 1].hi - self.Nodes[(ind << 1) + 1].lo + 1)
        self.Nodes[(ind << 1) + 1].max = self.Nodes[ind].max
        self.Nodes[(ind << 1) + 1].min = self.Nodes[ind].min
        
        self.Nodes[(ind << 1) + 2].sum = self.Nodes[ind].max * (self.Nodes[(ind << 1) + 2].hi - self.Nodes[(ind << 1) + 2].lo + 1)
        self.Nodes[(ind << 1) + 2].max = self.Nodes[ind].max
        self.Nodes[(ind << 1) + 2].min = self.Nodes[ind].min
        
        self.lazy[ind] = False
        self.lazy[(ind << 1) + 1] = True
        self.lazy[(ind << 1) + 2] = True
    
    def update(self, val, l, r=-1):
        if r >= 0:
            self._update(0, l, r, val)
        else:
            self._update(0, l, l, val)
        
    def _update(self, ind, l, r, val):
        lo = self.Nodes[ind].lo
        hi = self.Nodes[ind].hi
        if r < lo or l > hi:
            return 
        if l <= lo and r >= hi:
            self.Nodes[ind].sum = val * (hi - lo + 1)
            self.Nodes[ind].min = val
            self.Nodes[ind].max = val
            self.lazy[ind] = True
        else:
            self.push_down(ind)
            self._update((ind << 1) + 1, l, r, val)
            self._update((ind << 1) + 2, l, r, val)
            self.push_up(ind)
    
    def _query(self, ind, l, r):
        lo = self.Nodes[ind].lo
        hi = self.Nodes[ind].hi
        if r < lo or l > hi:
            return 0, 10 ** 18, -10 ** 18
        if l <= lo and r >= hi:
            return self.Nodes[ind].sum, self.Nodes[ind].min, self.Nodes[ind].max
            
        self.push_down(ind)

        s1, mi1, ma1 = self._query((ind << 1) + 1, l, r)
        s2, mi2, ma2 = self._query((ind << 1) + 2, l, r)
        
        s = s1 + s2
        mi = min(mi1, mi2)
        ma = max(ma1, ma2)
            
        return s, mi, ma
    
    def query_max(self, l, r):
        return self._query(0, l, r)[2]
    
    def query_min(self, l, r):
        return self._query(0, l, r)[1]
    
    def query_sum(self, l, r):
        return self._query(0, l, r)[0]
    
    

    
            
class Segment_Tree():
    def __init__(self, arr):
        self.n = len(arr)
        logs = 1
        tmp = self.n - 1
        while tmp:
            logs += 1
            tmp >>= 1
        
        self.Nodes = [0] * ((1 << logs) - 1)
        self.lazy = [None] * ((1 << logs) - 1)
        self.los = [0] * ((1 << logs) - 1)
        self.his = [0] * ((1 << logs) - 1)
        
        self.arr = arr
        
        
        
        self.build(0, 0, self.n - 1)
        
    def build(self, ind, lo, hi):
        if lo == hi:
            val = self.arr[lo]
            self.Nodes[ind] = val
            return
    
        mid = (lo + hi) // 2
        left = (ind << 1) + 1
        right = (ind << 1) + 2
        self.build(left, lo, mid)
        self.build(right, mid + 1, hi)
        self.Nodes[ind] = self.Nodes[left] + self.Nodes[right]
        return
    def push_up(self, ind):
        self.Nodes[ind] = self.Nodes[(ind << 1) + 1] + self.Nodes[(ind << 1) + 2]
        
    def push_down(self, ind):
        if self.lazy[ind] == None: return
        self.Nodes[(ind << 1) + 1] = self.lazy[ind] * (self.his[(ind << 1) + 1] - self.los[(ind << 1) + 1] + 1)
        self.Nodes[(ind << 1) + 2] = self.lazy[ind] * (self.his[(ind << 1) + 2] - self.los[(ind << 1) + 2] + 1)
        self.lazy[ind] = None
    
    def update(self, val, l, r=-1):
        if r >= 0:
            self._update(0, l, r, val)
        else:
            self._update(0, l, l, val)
        
    def _update(self, ind, l, r, val):
        lo = self.los[ind]
        hi = self.his[ind]
        if r < lo or l > hi:
            return 
        if l <= lo and r >= hi:
            self.Nodes[ind].sum = val * (hi - lo + 1)
            self.lazy[ind] = val
        else:
            self.push_down(ind)
            self._update((ind << 1) + 1, l, r, val)
            self._update((ind << 1) + 2, l, r, val)
            self.push_up(ind)
    
    def _query(self, ind, l, r):
        lo = self.los[ind]
        hi = self.his[ind]
        if r < lo or l > hi:
            return 0
        if l <= lo and r >= hi:
            return self.Nodes[ind]
            
        self.push_down(ind)

        s1 = self._query((ind << 1) + 1, l, r)
        s2 = self._query((ind << 1) + 2, l, r)
            
        return s1 + s2
    
    def query_sum(self, l, r):
        return self._query(0, l, r)
        
            
            
        
        
        
            
        
        
        

File: Templates.cpp
#include <tuple>
#include <iostream>
#include <vector>
#include <string>
#include <queue>
#include <cmath>
#include <algorithm>
#include <unordered_map>
#include <unordered_set>

#define ll long long
#define pii pair<int, int>

using namespace std;

// int main() {
//     cout.sync_with_stdio(false);
//     cin.tie(nullptr);
//     int t;
//     cin >> t;
//     while (t--) {
//         int n, k, res;
//         ll l;
//         char c;
//         string s;
//         cin >> n;
//         vector<int> arr(n);
//         for (int i = 0; i < n; ++i) cin >> arr[i];

//         cout << res << endl;
//     }
//     return 0;
// }


// Binary Index Tree
template<typename valueType>
class BIT {
private:
    vector<valueType> nums;
    vector<valueType> sums;
    valueType n;
public:
    BIT(valueType len) {
        n = len;
        nums = vector<valueType> (len + 1, 0);
        sums = vector<valueType> (len + 1, 0);
    }

    bool Update(valueType ind, valueType val) {
        if (ind < 0 || ind >= n) return false;
        ind++;
        valueType diff = nums[ind] - val;
        nums[ind] = val;
        while (ind <= n) {
            sums[ind] -= diff;
            ind += (ind & -ind);
        }
        return true;
    }

    valueType Query(valueType ind) {
        if (ind < 0 || ind >= n) return 0;
        ind++;
        valueType sum = 0;
        while (ind > 0) {
            sum += sums[ind];
            ind -= (ind & -ind);
        }
        return sum;
    }
};


// Disjoint Set
template<typename valueType>
class DJS {
private:
    vector<valueType> arr;
    vector<valueType> rank;

public:
    DJS(valueType len) {
        arr = vector<valueType> (len);
        for (int i = 0; i < len; ++i) arr[i] = i;
        rank = vector<valueType> (len, 1);
    }

    valueType Find(valueType x) {
        if (arr[x] != x)
            arr[x] = find(arr[x]);
        return arr[x];
    }

    valueType Union(valueType x, valueType y) {
        valueType xp = find(x);
        valueType yp = find(y);
        if (xp != yp) {
            if (rank[xp] >= rank[yp]) {
                arr[yp] = xp;
                rank[xp] += rank[yp];
            } else {
                arr[xp] = yp;
                rank[yp] += rank[xp];
            }
        }
    }
};


// Segment Tree   
template<typename valueType>     
struct Segment_TreeNode {
    int lo, hi;
    valueType sum, min, max;
    Segment_TreeNode(int lo, int hi, valueType s, valueType mi, valueType ma): lo(lo), hi(hi), sum(s), min(mi), max(ma) {};
};

template<typename valueType>         
class Segment_Tree {
private:
    vector< Segment_TreeNode<valueType>* > nodes;
    vector<bool> lazy;
    int n;
    vector<valueType> arr;

    void build(int ind, int lo, int hi) {
        if (lo == hi) {
            valueType val = arr[lo];
            nodes[ind] = new Segment_TreeNode<valueType> (lo, hi, val, val, val);
            return;
        }
        int mid = (lo + hi) / 2;
        int left = (ind << 1) + 1;
        int right = (ind << 1) + 2;

        build(left, lo, mid);
        build(right, mid + 1, hi);

        valueType nSum, nMin, nMax;
        nSum = nodes[left]->sum + nodes[right]->sum;
        nMin = min(nodes[left]->min, nodes[right]->min);
        nMax = max(nodes[left]->max, nodes[right]->max);
        
        nodes[ind] = new Segment_TreeNode<valueType> (lo, hi, nSum, nMin, nMax);
    }
    void push_up(int ind) {
        nodes[ind]->sum = nodes[(ind << 1) + 1]->sum + nodes[(ind << 1) + 2]->sum;
        nodes[ind]->min = min(nodes[(ind << 1) + 1]->min, nodes[(ind << 1) + 2]->min);
        nodes[ind]->max = max(nodes[(ind << 1) + 1]->max, nodes[(ind << 1) + 2]->max);
    }
    void push_down(int ind) {
        if (!lazy[ind]) return;
        nodes[(ind << 1) + 1]->sum = nodes[ind]->max * (nodes[(ind << 1) + 1]->hi - nodes[(ind << 1) + 1]->lo + 1);
        nodes[(ind << 1) + 1]->max = nodes[ind]->max;
        nodes[(ind << 1) + 1]->min = nodes[ind]->min;
        
        nodes[(ind << 1) + 2]->sum = nodes[ind]->max * (nodes[(ind << 1) + 2]->hi - nodes[(ind << 1) + 2]->lo + 1);
        nodes[(ind << 1) + 2]->max = nodes[ind]->max;
        nodes[(ind << 1) + 2]->min = nodes[ind]->min;
        
        lazy[ind] = false;
        lazy[(ind << 1) + 1] = true;
        lazy[(ind << 1) + 2] = true;
    }
    void _update(int ind,int l,int r,valueType val) {
        int lo = nodes[ind]->lo;
        int hi = nodes[ind]->hi;
        if (r < lo || l > hi) return ;
        if (l <= lo && r >= hi) {
            nodes[ind]->sum = val * (hi - lo + 1);
            nodes[ind]->min = val;
            nodes[ind]->max = val;
            lazy[ind] = true;
        } else {
            push_down(ind);
            _update((ind << 1) + 1, l, r, val);
            _update((ind << 1) + 2, l, r, val);
            push_up(ind);
        }
    }
public:
    Segment_Tree(int num) {
        n = num;
        nodes.resize(4*n);
        lazy.resize(4*n);
        arr.resize(n, 0);
        build(0, 0, n - 1);
    }
    Segment_Tree(vector<valueType> _arr) {
        n = _arr.size();
        nodes.resize(4*n);
        lazy.resize(4*n);
        arr = _arr;
        build(0, 0, n - 1);
    }

    void update(int l,int r,valueType val) {
        _update(0, l, r, val);
    }

    tuple<valueType, valueType, valueType> _query(int ind,int l,int r) {
        int lo = nodes[ind]->lo;
        int hi = nodes[ind]->hi;
        if (r < lo || l > hi)
            return make_tuple(0, numeric_limits<valueType>::max(), numeric_limits<valueType>::min());
        if (l <= lo && r >= hi)
            return make_tuple(nodes[ind]->sum, nodes[ind]->min, nodes[ind]->max);
            
        push_down(ind);

        valueType s1, mi1, ma1, s2, mi2, ma2, s, mi, ma;
        tie(s1, mi1, ma1) = _query((ind << 1) + 1, l, r);
        tie(s2, mi2, ma2) = _query((ind << 1) + 2, l, r);
        
        s = s1 + s2;
        mi = min(mi1, mi2);
        ma = max(ma1, ma2);
            
        return make_tuple(s, mi, ma);
    }
    int myquery(int tar,int  ind=0) {
        int lo = nodes[ind]->lo;
        int hi = nodes[ind]->hi;
        if (nodes[ind]->min > tar) return -1;
        else if (lo == hi) return lo;
        push_down(ind);
        int tmp = myquery(tar, (ind << 1) + 1);
        if (tmp == -1)
            tmp = myquery(tar, (ind << 1) + 2);
        return tmp;
    }

    valueType query_max(int l,int r) {return get<2>(_query(0, l, r));}
    valueType query_min(int l,int r) {return get<1>(_query(0, l, r));}
    valueType query_sum(int l,int r) {return get<0>(_query(0, l, r));}

};